test/integration/volume/persistent_volumes_test.go

```diff
@@ -1106,6 +1266,27 @@ func waitForAnyPersistentVolumeClaimPhase(w watch.Interface, phase v1.Persistent
 	}
 }
 
+func waitForPersistentVolumeClaimStorageClass(client *clientset.Clientset, claimName, namespace, scName string, w watch.Interface) {
+	// Check if the claim already has the storage class
+	claim, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(context.TODO(), claimName, metav1.GetOptions{})
+	if err == nil && *claim.Spec.StorageClassName == scName {
+		return
+	}
+
+	// Wait for the storage class
+	for {
```

Comment:
Put a timeout on this? xref `apimachinery/watch.go:waitForEvent`

Reply:
Removed for now to pass feature freeze, integration tests will be added during this week as it needs further debugging.

Reply:
@jpbetz Added [here](https://github.com/kubernetes/kubernetes/pull/111746/files#diff-6d10cbc48ef971686deb638dbc66a7f0498de8d9480f5de1275bbde421d379a5R1303)